cpp/test/sparse/neighbors/connect_components.cu

```diff
@@ -353,5 +360,612 @@ TEST_P(ConnectComponentsTestF_Int, Result)
 INSTANTIATE_TEST_CASE_P(ConnectComponentsTest,
                         ConnectComponentsTestF_Int,
                         ::testing::ValuesIn(fix_conn_inputsf2));
+
+template <typename value_idx, typename value_t>
+struct MutualReachabilityFixConnectivitiesRedOp {
```

Comment:
General comment:
I am assuming that `MutualReachabilityFixConnectivitiesRedOp` is copied over from cuML. It is good that we are testing here that it works. It does, however, suggest to me that cuML and RAFT are too tightly coupled in this case. Perhaps in a future PR, we can move this functionality into RAFT to reduce our API surface/coupling?

Reply:
Yes, that can be done in the future. I can create an issue for the same.

Reply:
Yeah, let's make an issue for this for sure. I think the whole mutual reachability primitive can be extracted from hdbscan and moved over to RAFT, including potentially the computations of core distances (distances to the kth neighbor). Specifically, the mutual reachability ends up requiring changes to each of the knn/ann algorithms to support an `epilogue` functor so we could isolate/instantiate these in RAFT and expose to cuML

Reply:
Created https://github.com/rapidsai/raft/issues/1595